### ISSUE TITLE
A is not associated to B

### ISSUE THREAD

**Issue by sadeghipour** [sequelize/sequelize-auto#369]
Hi,

I generated all models by sequelize-auto but when I tried to make an inner join on A and B table I received an error:
A is not associated to B

I read all related posts and finally below code block works:

A.hasMany(Post, {foreignKey: 'user_id'})
B.belongsTo(User, {foreignKey: 'user_id'})

B.find({ where: { ...}, include: [A]})  

But I think it horrible to define it on every query.

Is it any way to generate it automatically with sequelize-auto in model?

Thanks for your help.

**Comment by steveschmitt** [comment]
Added `belongsTo/hasOne/hasMany` to the generated `initModels` function in release **0.7.0**.

**Comment by sadeghipour** [comment]
Hi @steveschmitt,
Thanks for your update.
Unfortunately there is an error on generating initModels.

for example:
imagine that we have 2 tables that names are **appointments** and **branches**
and appointments has a foreign key to branches (**appointments.branch_id = branch.id**) 
but when initModels generated its connected reverse to each other:
**the query of** 

```
Appointments.findOne({where:{id:1},include: [{
      model: Branches,
      attributes:["id","name"]
    }]})
```

**equal to** 

SELECT `appointments`.`id`, `appointments`.`branch_id`, `branch`.`id` AS `branch.id`, `branch`.`name` AS `branch.name` FROM `appointments` AS `appointments` LEFT OUTER JOIN `branches` AS `branch` **ON `appointments`.`id` = `branch`.`id` WHERE `appointments`.`id` = 1;**


initModels looks like:

```
var DataTypes = require("sequelize").DataTypes;
var _appointments = require("./appointments");
var _branches = require("./branches");

function initModels(sequelize) {
  var appointments = _appointments(sequelize, DataTypes);
  var branches = _branches(sequelize, DataTypes);

  //Problem might be in here: 
  appointments.belongsTo(branches, { foreignKey: "id"});
  branches.hasMany(appointments, { foreignKey: "branch_id"});
  
  return {
    appointments,
    branches,
  };
}
module.exports = initModels;
module.exports.initModels = initModels;
module.exports.default = initModels;
```

**if we change initModels as below**
appointments.belongsTo(branches **, { foreignKey: "branch_id"}** );
  
branches.hasMany(appointments **, { foreignKey: "id"}** );

 **sql query will:**
 SELECT `appointments`.`id`, `appointments`.`branch_id`, `branch`.`id` AS `branch.id`, `branch`.`name` AS `branch.name` FROM `appointments` AS `appointments` LEFT OUTER JOIN `branches` AS `branch` ON **`appointments`.`branch_id` = `branch`.`id`** WHERE `appointments`.`id` = 1;

**Comment by steveschmitt** [comment]
You are right - I only tested it on one direction.  I think the initModels needs to have `branch_id` in both the `belongsTo` and `hasMany` statements: 
```js
  appointments.belongsTo(branches, { foreignKey: "branch_id"});
  branches.hasMany(appointments, { foreignKey: "branch_id"});
```

**Comment by steveschmitt** [comment]
Fixed in 0.7.1
